src/theme/themeMerge/themeMerge.ts

```diff
@@ -0,0 +1,69 @@
+/*
+    Utility method to help a user merge their settings into 'RootTheme'. Returns a new object
+    (deep clone).
+
+    ex: const MyTheme = themeMerge({
+        fonts: {
+            base: 'Comic Sans MS',
+        },
+    });
+
+    TODO: It would be good at some point to also add validation to this so the user doesn't break
+    all the things.
+*/
+
+type ThemeType = {
+    typography: object;
+    radii: object;
+    buttons: object;
+    inputs: object;
+    colors: object;
+    borders: object;
+    skeleton: object;
+    [key: string]: any;
+};
+
+// Can't really be 100% sure what data will be as it is cloned, hence :any
+function deepClone(data: any) {
```

Comment:
Can we get at least 1 test for this function, to verify that mutating the clone doesn't mutate the original?

Reply:
This can be accomplished by testing `themeMerge` itself, I neglected to write one though which was dumb.